src/drivers/nuopc/cime_driver/esmApp.F90

```diff
@@ -10,23 +10,33 @@ program esmApp
   use ESMF, only             : ESMF_GridCompDestroy, ESMF_LOGMSG_INFO, ESMF_GridComp, ESMF_GridCompRun
   use ESMF, only             : ESMF_GridCompFinalize, ESMF_GridCompCreate, ESMF_GridCompInitialize
   use ESMF, only             : ESMF_LOGKIND_MULTI_ON_ERROR
+  use mpi, only              : MPI_COMM_WORLD, MPI_COMM_NULL, MPI_Init, MPI_FINALIZE
   use NUOPC, only            : NUOPC_FieldDictionarySetup
   use ensemble_driver,  only : SetServices
-
+  use shr_pio_mod,      only : shr_pio_init1, shr_pio_init2
   implicit none
-
+  integer                    :: COMP_COMM
   integer                    :: rc, urc
   type(ESMF_GridComp)        :: ensemble_driver_comp
 
+  call MPI_init(rc)
+  COMP_COMM = MPI_COMM_WORLD
+  ! This call must be after initializing mpi but before initializing esmf, so that io tasks can be
+  ! spun up if pio_async is true
```

Comment:
Can you add some more comments:

Reply:
8 is the number of components which is currently hard-wired and COMP_COMM is returned from drv_in
all shr_pio_init1(8, "drv_in", COMP_COMM) - if async is false the COMP_COMM will be MPI_COMM_WORLD - if async is true then COMP_COMM will be MPI_COMM_WORLD minus the io tasks